examples/ex6p.cpp

```diff
@@ -220,17 +224,26 @@ int main(int argc, char *argv[])
 
       // 17. Solve the linear system A X = B.
       //     * With full assembly, use the BoomerAMG preconditioner from hypre.
-      //     * With partial assembly, use no preconditioner, for now.
-      HypreBoomerAMG *amg = NULL;
-      if (!pa) { amg = new HypreBoomerAMG; amg->SetPrintLevel(0); }
+      //     * With partial assembly, use the diagonal preconditioner.
+      Solver *M = NULL;
+      if (pa)
+      {
+         M = new OperatorJacobiSmoother(a, ess_tdof_list);
+      }
+      else
+      {
+         HypreBoomerAMG *amg = new HypreBoomerAMG;
+         amg->SetPrintLevel(0);
+         M = amg;
```

Comment:
The `amg` variable is now unneccessary here...

Reply:
To set the print level of the `HypreBoomerAMG` solver via `SetPrintLevel`, I would then have to use
```
M = new HypreBoomerAMG;
static_cast<HypreBoomerAMG*>(M)->SetPrintLevel(0);
```
Personally, I think using variable `amg` makes it easier to read but I can change it if casting this way is preferred.

Reply:
Sorry, I overlooked the fact that the pointers are not the same type. Please disregard.